chain-impl-mockchain/src/evm/mod.rs

```diff
@@ -216,6 +244,48 @@ impl Readable for EvmTransaction {
                     access_list,
                 })
             }
+            #[cfg(feature = "evm")]
+            1 => {
+                // CREATE2 Transaction
+                let caller = read_address(buf)?;
+                let value = read_u256(buf)?;
+                let init_code = read_bytecode(buf)?;
+                let salt = read_h256(buf)?;
+                let gas_limit = read_gas_limit(buf)?;
+                let access_list = read_access_list(buf)?;
+
+                buf.expect_end()?;
```

Comment:
I see this now, but I don't think this works in settings other than just the simple struct ser/de where an `EvmTransaction` is not the only element in the `ReadBuf`.
For instance, in the transaction format this is likely not the last thing in the buffer, nor we should limit binary layout because of this.